Scripts/Mobiles/PlayerMobile.cs

```diff
@@ -3755,6 +3758,14 @@ public override bool OnBeforeDeath()
             if (Criminal)
                 BuffInfo.RemoveBuff(this, BuffIcon.CriminalStatus);
 
+            if (Backpack != null)
+            {
+                GemOfSalvation gem = Backpack.FindItemByType<GemOfSalvation>();
+
+                if (gem != null)
+                    Timer.DelayCall(TimeSpan.FromSeconds(2.0), new TimerStateCallback<PlayerMobile>(gem.Use), this);
+            }
+
```

Comment:
Can we invoke an event for this (PlayerDeathEvent)? Also, should we be using Persistence to serialize all these things that very few players will use, but will cause memory usage on all instances?